FastSimulation/Tracking/interface/SeedFinderSelector.h

```diff
@@ -32,16 +44,114 @@ class SeedFinderSelector
     void setTrackingRegion(const TrackingRegion * trackingRegion){trackingRegion_ = trackingRegion;}
     
     bool pass(const std::vector<const FastTrackerRecHit *>& hits) const;
+    std::string LayerName(int layerN, std::string layerside, bool IsPixBarrel) const;    
 
 private:
     
     std::unique_ptr<HitTripletGeneratorFromPairAndLayers> pixelTripletGenerator_;
     std::unique_ptr<MultiHitGeneratorFromPairAndLayers> multiHitGenerator_;
+    std::unique_ptr<CAHitTripletGenerator> CAHitTriplGenerator_;
+    std::unique_ptr<CAHitQuadrupletGenerator> CAHitQuadGenerator_;
     const TrackingRegion * trackingRegion_;
     const edm::EventSetup * eventSetup_;
     const MeasurementTracker * measurementTracker_;
     const std::string measurementTrackerLabel_;
-    
+    SeedingLayerSetsBuilder* seedingLayers_;
+    std::unique_ptr<SeedingLayerSetsHits> seedingLayer;
+    std::vector<unsigned> layerPairs_;
+    edm::ESHandle<TrackerTopology> trackerTopology;
+    std::unique_ptr<IHD::ImplBase> impl_;
+    edm::Event * event_;
 };
 
+namespace IHD {
```

Comment:
You should not need any of this stuff as there is only a single construction of `impl_`. The complexity is needed in `HitPairEDProducer` because that EDProducer has to support multitude of cases.

Reply:
Right... thanks. I have removed all of this now.